### PR TITLE
fix k8s image publish

### PR DESCRIPTION
**Pull request by bengl**
Fixes the k8s image publish by using the github token and using the tag everywhere.

The following patch was used to test it.

 [03c708aed26c836af0087141290f3caaab3c61fa](https://gist.github.com/bengl/d1f95e79f237595bb52301f1f2241697)
 
 **Note to reviewers:** please verify that the patch above makes sense for testing this.

### CODE REVIEW

File: k8s/auto-inst/Dockerfile [hunk 1]
```diff
@@ -7,4 +7,6 @@ COPY . .
 
 RUN npm install ./dd-trace.tgz
 
+FROM busybox
+
```
Comment (ahmed-mez) [inline k8s/auto-inst/Dockerfile:11]: We need to copy the library from the build image to the new one `COPY --from=build <src> <dst>`

**Review by Qard — DISMISSED** [review]

File: k8s/auto-inst/Dockerfile (hunk 1 above)
Comment (bengl) [reply k8s/auto-inst/Dockerfile:11]: oh gotcha. will do.

**Review by Qard — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -30,6 +30,24 @@ deploy_to_docker_registries:
     branch: main
     strategy: depend
   variables:
-    IMG_SOURCES: ghcr.io/datadog/dd-trace-js/dd-lib-js-init:$CI_COMMIT_SHA
+    IMG_SOURCES: ghcr.io/datadog/dd-trace-js/dd-lib-js-init:$CI_COMMIT_TAG
```
Comment (Kyle-Verhoog) [inline .gitlab-ci.yml:33]: java uses the SHA here: https://github.com/DataDog/dd-trace-java/blob/master/.gitlab-ci.yml#L136 is it different for node?

**Review by Kyle-Verhoog — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (Kyle-Verhoog) [reply .gitlab-ci.yml:33]: ah yeah, seems to line up with the tag that is published: https://github.com/DataDog/dd-trace-js/pull/2340/files#diff-57b402748df77503e37b3b4f47152b3cfb6502f3d2d00c79d4512a77df802333R55

Comment (bengl) [reply .gitlab-ci.yml:33]: Yeah since we're publishing the tag explicitly.

**Pull request merged**